spec/models/jobseeker_profile_spec.rb

```diff
@@ -22,6 +22,26 @@
         expect(profile.qualified_teacher_status_year).to eq(previous_application.qualified_teacher_status_year)
         expect(profile.qualified_teacher_status).to eq(previous_application.qualified_teacher_status)
         expect(profile.personal_details.first_name).to eq(previous_application.first_name)
+        expect(profile.personal_details.last_name).to eq(previous_application.last_name)
+      end
+
+      it "creates a job preferences record" do
+        expect(profile.job_preferences).to be_present
+      end
+    end
+
+    context "when the jobseeker has a previous draft application" do
+      before do
+        create(:job_application, :status_draft, jobseeker:)
+      end
+
+      it "does not use the details from the draft application" do
+        expect(profile.employments.map(&:job_title)).to be_empty
+        expect(profile.qualifications.map(&:institution)).to be_empty
+        expect(profile.qualified_teacher_status_year).to be_nil
+        expect(profile.qualified_teacher_status).to be_nil
+        expect(profile.personal_details.first_name).to be_nil
+        expect(profile.personal_details.last_name).to be_nil
```

Comment:
this is a lot of negative asserts - presumably e.g. the employment list is empty, so the map() call does nothing useful. Is there a way of making this a positive test about what it **does** do (which I don't know BTW, but would like to)

Reply:
Removed the map, good call! It creates empty models so I am not sure there is much more we can do